grakn-core/src/main/java/ai/grakn/graql/admin/ReasonerQuery.java

```diff
@@ -91,9 +91,10 @@
     /**
      * resolves the query
      * @param materialise materialisation flag
+     * @param explanation whether to provide explanation
      * @return stream of answers
      */
-    Stream<Answer> resolve(boolean materialise);
+    Stream<Answer> resolve(boolean materialise, boolean explanation);
```

Comment:
This is really just for my own understanding: 

1. I am assuming you cannot retroactively get the explanation for something that was resolved? I.e. if something is materialised then you can't get the explanation?

2. Does computing the explanation affect performance?